framework/external/easylogging/easylogging++.cc

```diff
@@ -2605,7 +2605,12 @@ Writer& Writer::construct(int count, const char* loggerIds, ...) {
 
 void Writer::initializeLogger(const std::string& loggerId, bool lookup, bool needLock) {
   if (lookup) {
-    m_logger = ELPP->registeredLoggers()->get(loggerId, ELPP->hasFlag(LoggingFlag::CreateLoggerAutomatically));
+    auto thread_logger_id = ELPP->getThreadLoggerId();
+    if (thread_logger_id.empty()) {
+      m_logger = ELPP->registeredLoggers()->get(loggerId, ELPP->hasFlag(LoggingFlag::CreateLoggerAutomatically));
+    } else {
+      m_logger = ELPP->registeredLoggers()->get(thread_logger_id, ELPP->hasFlag(LoggingFlag::CreateLoggerAutomatically));
+    }
```

Comment:
```suggestion
    auto logger_id = ELPP->getThreadLoggerId().empty() ? loggerId : ELPP->getThreadLoggerId();
    m_logger = ELPP->registeredLoggers()->get(logger_id, ELPP->hasFlag(LoggingFlag::CreateLoggerAutomatically));
```

Reply:
But, I don't think this change should be here.
You should pass the correct `loggerId` in the first place (from whoever initializes the `Writer` [constructor?]).
When is the `Writer` initialized anyway?

Reply:
I don't like your suggestion since defining a new variable with the same name as one of the function's argument is confusing. Also, the cppcheck will complain about it that it is a variable shadowing.

A `Writer` is an object that is constructed when calling to `LOG(LEVEL)` or `CLOG(LEVEL, ID)`

Reply:
So change the name of the variable.. It's not the point here.
So do you mean we don't have it on external libs? What external libs?

Reply:
External lib is not accurate. Just "lib".
Rember we talked about that If we want to write to specific logger ID we have to use `CLOG(LEVEL, ID)`.
So we did macro hack for that that could work on files of a specific process, but we had a problem with a print in a lib.
For example if there is a function on network_utils that print something. So in order to make it print to the right file, we would have to provide the longer ID to each function and change every LOG(LEVEL) to CLOG(LEVEL, ID).
So this is the solution that allows us not to provide this, and instead identifying it automatically.

Anyway, I narrowed the condition as you wanted.